lms/djangoapps/grades/api/views.py

```diff
@@ -143,13 +160,26 @@ def get(self, request, course_id):
                 error_code='user_mismatch'
             )
 
+        # Returns a 404 response if course_id is invalid, or request.user is not enrolled in the course
         course = self._get_course(course_id, request.user, 'load')
         if isinstance(course, Response):
+            # A student can only access her own grade status info
+            if request.user.username != username:
+                return user_mismatch_error()
```

Comment:
Why is it necessary to verify that the `username` matches here?  Isn't this already on the error path - with returning an error from `_get_course`?  Also, staff access isn't checked here, but is below.

If you really want the `username` verification to take precedence over the `course` verification, then simply move it above `_get_course`.  Otherwise, you can just delete these 3 lines.